resilience4j-retry/src/main/java/io/github/resilience4j/retry/RetryConfig.java

```diff
@@ -243,14 +280,19 @@ public Builder(RetryConfig baseConfig) {
         }
 
         public RetryConfig build() {
+            if (intervalFunction != null && intervalBiFunction != null) {
+                throw new IllegalStateException("The intervalFunction was configured twice which could result in an" +
+                    " undesired state. Please use either intervalFunction or intervalBiFunction.");
+            }
             RetryConfig config = new RetryConfig();
-            config.intervalFunction = intervalFunction;
+            config.intervalFunction = Optional.ofNullable(intervalFunction).orElse(IntervalFunction.ofDefaults());
```

Comment:
`config.intervalFunction` isn't used anymore in `RetryImpl`. 
If set only `IntervalFunction`, I think code should be like following 

```java
config.intervalBiFunction = Optional.ofNullable(intervalBiFunction)
                                    .orElse(IntervalBiFunction.ofIntervalFunction(intervalFunction));
```

Reply:
this means that `config.intervalFunction` should be marked as deprecated and nullable

Reply:
Yes, I think that's fine

Reply:
Hmm, I tried this and found an interesting use case, the following code will throw `IllegalStateException`

```java
RetryConfig baseConfig = RetryConfig.ofDefaults();
RetryConfig retryConfig = RetryConfig.from(baseConfig)
            .intervalFunction(IntervalFunction.of(100L))
            .build();
```

Because `baseConfig.intervalBiFunction = bi function of interval defaults` and since `config.intervalBiFunction` is the only used by `Retry`, I thought that only this value should be copied from base config on `RetryConfig.from(baseConfig)`, therefore `retryConfig.intervalBiFunction = baseConfig.intervalBiFunction`

I'm kind of stuck with this, any idea?

Reply:
some code
```java
private IntervalBiFunction<T> createIntervalBiFunction() {
            if (intervalFunction != null && intervalBiFunction != null) {
                throw new IllegalStateException("The intervalFunction was configured twice which could result in an" +
                    " undesired state. Please use either intervalFunction or intervalBiFunction.");
            }
            if (intervalFunction == null && intervalBiFunction == null) {
               return IntervalBiFunction.ofDefaults();
            }
            if (intervalFunction != null) {
                return IntervalBiFunction.ofIntervalFunction(intervalFunction);
            }

            return intervalBiFunction;
        }
```

```java
public Builder(RetryConfig baseConfig) {
            this.maxAttempts = baseConfig.maxAttempts;
            this.retryOnExceptionPredicate = baseConfig.retryOnExceptionPredicate;
            this.retryOnResultPredicate = baseConfig.retryOnResultPredicate;
            this.retryExceptions = baseConfig.retryExceptions;
            this.ignoreExceptions = baseConfig.ignoreExceptions;
            this.intervalBiFunction = baseConfig.intervalBiFunction;
        }
```

Reply:
Maybe we just use a int counter. If the counter == 2, then the user has used both methods.

Reply:
never mind, I think I was too tired

Reply:
This is how it was implemented in CircuitBreakerConfig now -> https://github.com/resilience4j/resilience4j/pull/1216/files

Reply:
I was trying to avoid that kind of implementation. 

What I've done is this: whenever `intervalFunction` is set then when creating a new builder with `RetryConfig.from` only `intervalFunction` is copied from base config. Because `intervalBiFunction` is always present and when `build` is invoked will be created from `intervalFunction`. 

See the implementation :)